util/bitstream.go

```diff
@@ -226,3 +256,22 @@ func ParseUInt16(data []byte, bitStartIndex uint16) (uint16, error) {
 	}
 	return binary.BigEndian.Uint16([]byte{leftByte, rightByte}), nil
 }
+
+func (bs *BitStream) ReadTwoBitField(numFields int) ([]byte, error) {
+	result := []byte{}
+
+	if numFields <= 0 {
+		return result, fmt.Errorf("numFields is invalid")
+	}
+
+	maxFields := numFields * 2
+	for i := 0; i < maxFields; i += 2 {
```

Comment:
I think `maxFields` was an idea you abandoned. Just use `numFields` and count by 1.

Reply:
`maxFields` usage removed and moved to using `numFields` and incrementing by 1.